### DISCUSSION TITLE
In which order are Xdt Transforms resolved?

### DISCUSSION THREAD

**Discussion by rubiasc** [NuGet/Home#11466] [Q&A]
In which order are Xdt Transforms resolved for multiple transitive restores?
From my understanding it is the same order as the dependency resolution which adds references to a flat package graph 
which is represented in the project.assets.json file.

I was unable to obtain information how packages on a given level with no further dependencies are sorted, is it alphabetically?
Is it possible to override or influence the order, for instance introduce a topological sorting?

Any clarification or further information regarding this is appreciated.

**Comment by dominoFire** [comment]
When working with `PackageReference` projects, XDT transformations are not supported. See https://docs.microsoft.com/nuget/consume-packages/migrate-packages-config-to-package-reference#xdt-transforms-are-not-applied-when-the-package-is-installed-after-the-upgrade

Then, `project.assets.json` cannot say anything regarding XDT transformations.

XDT transforms are supported in projects working with `packages.config`

NuGet makes no assumptions on how XDT transforms are made.

Per package, transforms are sorted alphabetically:
https://github.com/NuGet/NuGet.Client/blob/37d61e55a7e486fba2d224effde3ac7b775f35b8/src/NuGet.Core/NuGet.PackageManagement/Utility/MSBuildNuGetProjectSystemUtility.cs#L146
